pkg/configuration/config_manager/handler_util.go

```diff
@@ -189,13 +189,25 @@ func GetSupportReloadConfigSpecs(configSpecs []appsv1alpha1.ComponentConfigSpec,
 	return reloadConfigSpecMeta, nil
 }
 
-func FilterSubPathVolumeMount(metas []ConfigSpecMeta, volumes []corev1.VolumeMount) []ConfigSpecMeta {
+func FilterSupportReloadActionConfigSpecs(metas []ConfigSpecMeta, volumes []corev1.VolumeMount) []ConfigSpecMeta {
 	var filtered []ConfigSpecMeta
 	for _, meta := range metas {
 		v := FindVolumeMount(volumes, meta.ConfigSpec.VolumeName)
-		if v == nil || v.SubPath == "" || meta.ReloadType == appsv1beta1.TPLScriptType {
+		if isSyncReloadAction(meta.ConfigSpecInfo) || !isSubPathMount(v) {
```

Comment:
How about split this function into two?

The first one:

func FilterSubPathVolumeMount(metas []ConfigSpecMeta, volumes []corev1.VolumeMount) []ConfigSpecMeta {
    var filtered []ConfigSpecMeta
    for _, meta := range metas {
        v := FindVolumeMount(volumes, meta.ConfigSpec.VolumeName)
        if !isSubPathMount(v) {
            filtered = append(filtered, meta)
        }
    }
    return filtered
}

The second one:

// I'm not sure whether the name "FilterAsyncReloadAction" is proper
func FilterAsyncReloadAction(metas []ConfigSpecMeta) []ConfigSpecMeta {
    var filtered []ConfigSpecMeta
    for _, meta := range metas {
        if isSyncReloadAction(meta.ConfigSpecInfo){
            filtered = append(filtered, meta)
        }
    }
    return filtered
}

And two functions will be called in a chain:

	configSpecMetas = cfgcm.FilterSubPathVolumeMount(configSpecMetas, volumeDirs)
	configSpecMetas = cfgcm.FilterAsyncReloadAction(configSpecMetas)
	if len(configSpecMetas) == 0 {
		return nil
	}

I think that would be read much more clear